internal/app/game/game.go

```diff
@@ -0,0 +1,126 @@
+package game
+
+import (
+	"fmt"
+	"github.com/Piyushh1/mock-machine-coding-1/internal/pkg/utils"
+	"strconv"
+)
+
+var (
+	players []player
+	ladder = map[int]int{}
+	snake = map[int]int{}
+)
+
+const (
+	// should be configurable
+	finalCell = 100
+)
+
+func NewManager() Manager {
+	m := new(manager)
+
+	return m
+}
+
+type Manager interface {
+	InitLadder(startCell, endCell string)
+	InitSnakes(endCell, startCell string)
+	InitPlayers(name string)
+	StartGame()
+}
+
+type manager struct {
+}
+
+
+type player struct {
+	Name string
+	Position int
+}
+
+func (m *manager) InitLadder(startCell, endCell string) {
+	ec, _ := strconv.Atoi(endCell)
+	sc, _ := strconv.Atoi(startCell)
+	ladder[sc] = ec
+}
+
+func (m *manager) InitSnakes(endCell, startCell string) {
+	ec, _ := strconv.Atoi(endCell)
+	sc, _ := strconv.Atoi(startCell)
+	snake[ec] = sc
+}
+
+func (m *manager) InitPlayers(name string) {
+	players = append(players, player{Name: name, Position: 0})
+}
+
+func checkLadder(cell int) int {
+	newCell, ok := ladder[cell]
+	 if ok {
+	 	return checkLadder(newCell)
```

Comment:
should check for snake as well?

Reply:
wouldn't this create an infinite loop? I assumed a cell can't have ladder and snake both

Reply:
A snake can land you at a cell which may have a ladder.
A cell won't have start point of both snake and ladder but end point of one may be start point of another